src/spreadinterp.cpp

```diff
@@ -164,7 +164,7 @@ constexpr std::array<std::array<T, PaddedM>, N> pad_2D_array_with_zeros(
     const std::array<std::array<T, M>, N> &input) noexcept {
   constexpr auto pad_with_zeros = [](const auto &input) constexpr noexcept {
     std::array<T, PaddedM> padded{0};
-    for (auto i = 0; i < input.size(); ++i) {
+    for (size_t i = 0; i < input.size(); ++i) {
```

Comment:
I changed a lot of loop headers to silence compiler warnings.
The problem with
`for (auto i = 0; i < input.size(); ++i)`
is that `0` is an `int`; so `i` becomes `int`, but we really want it to be the type of `input.size()`.